src/main/service/sjpFilterService.ts

```diff
@@ -104,23 +121,48 @@ export class SjpFilterService {
         return activeFilters;
     }
 
-    private doFiltering(allCases, postcodeFilters, prosecutorFilters) {
+    private doFiltering(allCases, londonPostalAreaCodes, postcodeFilters, prosecutorFilters) {
         const filteredCases = [];
         allCases.forEach(item => {
             const formattedPostcode = item.postcode.split(' ', 2)[0];
+            const postalAreaCode = item.postcode.split(/\d/)[0];
             const formattedProsecutor = item.organisationName.replace(replaceRegex, '');
 
             if (postcodeFilters.length > 0 && prosecutorFilters.length > 0) {
                 if (postcodeFilters.includes(formattedPostcode) && prosecutorFilters.includes(formattedProsecutor)) {
                     filteredCases.push(item);
+                } else if (
+                    postcodeFilters.includes('The City of London') &&
+                    londonPostalAreaCodes.includes(postalAreaCode) &&
+                    prosecutorFilters.includes(formattedProsecutor)
+                ) {
+                    filteredCases.push(item);
+                }
+            } else if (postcodeFilters.length > 0) {
+                if (postcodeFilters.includes(formattedPostcode)) {
+                    filteredCases.push(item);
+                } else if (
+                    postcodeFilters.includes('The City of London') &&
+                    londonPostalAreaCodes.includes(postalAreaCode)
+                ) {
+                    filteredCases.push(item);
                 }
-            } else if (postcodeFilters.length > 0 && postcodeFilters.includes(formattedPostcode)) {
-                filteredCases.push(item);
             } else if (prosecutorFilters.length > 0 && prosecutorFilters.includes(formattedProsecutor)) {
                 filteredCases.push(item);
             }
         });
 
         return filteredCases;
     }
+
+    /**
+     * This method checks whether any of the cases have a postal code prefix that belongs to London.
+     * @param londonPostalAreaCodes The list of postal code prefixes for London.
+     * @param postalAreaCodes The list of postal code prefixes from the cases.
+     * @private
+     */
+    private checkForLondonPostalAreaCodes(londonPostalAreaCodes, postalAreaCodes) {
+        const postalAreaInLondon = new Set([...londonPostalAreaCodes].filter(element => postalAreaCodes.has(element)));
+        return postalAreaInLondon.size > 0;
+    }
```

Comment:
Think you can simply use  postalAreaCodes.some(p => londonPostalAreaCodes.includes(p)) ...that is a bit cleaner